controllers/authController.js

```diff
@@ -0,0 +1,227 @@
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    const avatarURL = gravatar.url(email, { s: "250", r: "pg", d: "mm" });
+
+    const newUser = new User({
+      email,
+      password: hashedPassword,
+      subscription: "starter",
+      avatarURL,
+      verificationToken,
+    });
+
+    await newUser.save();
+
+    await sendVerificationEmail(email, verificationToken);
+
+    res.status(201).json({
+      user: {
+        email: newUser.email,
+        subscription: newUser.subscription,
+        avatarURL: newUser.avatarURL,
+      },
+      token: jwt.sign({ id: newUser._id }, process.env.USER_TOKEN, {
+        expiresIn: "1h",
+      }),
+    });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Internal server error" });
+  }
+};
+
+const uploadAvatar = async (req, res) => {
+  try {
+    const { file } = req;
+    if (!file) {
+      return res.status(400).json({ message: "No file uploaded" });
+    }
+
+    const image = await jimp.read(file.path);
+    await image.cover(250, 250).write(file.path);
+
+    const newAvatarName = `${req.user._id}${path.parse(file.originalname).ext}`;
+    const newAvatarPath = path.join(
+      __dirname,
+      "../public/avatars",
+      newAvatarName
+    );
+    await fs.rename(file.path, newAvatarPath);
+
+    await User.findByIdAndUpdate(req.user._id, {
+      avatarURL: `/avatars/${newAvatarName}`,
+    });
+
+    res.json({ avatarURL: `/avatars/${newAvatarName}` });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Server error" });
+  }
+};
+
+const loginUser = async (req, res) => {
+  const { error } = loginSchema.validate(req.body);
+  if (error) {
+    return res.status(400).json({ message: error.details[0].message });
+  }
+
+  const { email, password } = req.body;
+
+  try {
+    const user = await User.findOne({ email });
+
+    if (!user) {
+      return res.status(401).json({ message: "Invalid credentials" });
+    }
+
+    const isPasswordValid = await bcrypt.compare(password, user.password);
+    if (!isPasswordValid) {
+      return res.status(401).json({ message: "Invalid credentials" });
+    }
+
+    if (!user.verify) {
+      return res.status(401).json({ message: "Email not verified" });
+    }
+
+    res.json({
+      user: {
+        email: user.email,
+        subscription: user.subscription,
+        avatarURL: user.avatarURL,
+      },
+      token: jwt.sign({ id: user._id }, process.env.USER_TOKEN, {
+        expiresIn: "1h",
+      }),
+    });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Internal server error" });
+  }
+};
+
+const verifyEmail = async (req, res) => {
+  const { verificationToken } = req.params;
+
+  try {
+    const user = await User.findOne({ verificationToken });
+
+    if (!user) {
+      return res.status(404).json({ message: "User not found" });
+    }
+
+    user.verify = true;
+    user.verificationToken = null;
```

Comment:
you could just do this
await User.findByIdAndUpdate(user._id, {
    verify: true,
    verificationToken: null,
  });